### DISCUSSION TITLE
PDF returns no text, tables, or anything

### DISCUSSION THREAD

**Discussion by toakleyy** [jsvine/pdfplumber#717] [Ask for help with specific PDFs]
I have used pdfplumber with success on a separate PDF, but now when I try the same function on this new PDF I get absolutely nothing. I'm simply trying to extract text.  See attached PDF here (I have blocked out personal info, but it is not blocked out when I try to plumb it) - 
[RJ PDF markup.pdf](https://github.com/jsvine/pdfplumber/files/9417718/RJ.PDF.markup.pdf)

Some code I am using that isn't working:
`
```
import pdfplumber
    
file = "localfilename"

with pdfplumber.open(file) as _pdf:
    pages = _pdf.pages

    for _i_, page in enumerate(pages):
        text = page.extract_text()
        print(f'page {_i_}')
        print(text)`
```

My output looks like this:
page 0

page 1

page 2

So it scans each page, but doesn't actually come up with any text.

Any help would be much appreciated. Thank you.

**Comment by jsvine** [comment]
Hi @toakleyy, and thanks for your interest in this library. From what you've shared, it appears that your PDF is a scanned document, not a born-digital PDF. (A quick way to test this: Can you select / copy / paste any text within the document?) If that's the case, you'll first want to run [optical character recognition](https://en.wikipedia.org/wiki/Optical_character_recognition) on the PDF. Then you can extract the OCR-detected text with `pdfplumber`.

**Reply by toakleyy** [reply]
Thanks, I will try this. Can you recommend an OCR library for Python?

**Reply by jsvine** [reply]
The most established open-source OCR tool is probably [Tesseract](https://tesseract-ocr.github.io/), though its quality can be a bit spotty. But certainly worth trying as a first option. [OCRMyPDF](https://ocrmypdf.readthedocs.io/en/latest/) is a Tesseract-based Python project that works both via the command line and as a Python library. I've used it with success in the past. Another Python- and Tesseract-based option is [PyTesseract library](https://pypi.org/project/pytesseract/), which I've also found to work well.
